frontend/src/components/c-summary-charts.vue

```diff
@@ -610,14 +628,25 @@ export default defineComponent({
 
       // skip if accidentally clicked on ramp chart
       if (this.drags.length === 2 && this.drags[1] - this.drags[0]) {
-        // additional day was added to include the date represented by filterUntilDate
-        const tdiff = (new Date(this.filterUntilDate)).valueOf() - (new Date(this.filterSinceDate)).valueOf()
-            + window.DAY_IN_MS;
-        const idxs = this.drags.map((x) => (x * tdiff) / 100);
-        const tsince = window.getDateStr(new Date(this.filterSinceDate).getTime() + idxs[0]);
-        const tuntil = window.getDateStr(new Date(this.filterSinceDate).getTime() + idxs[1]);
-        this.drags = [];
-        this.openTabZoom(user, tsince, tuntil, isMerged);
+        const optimisedMinimumDate = this.getOptimisedMinimumDate(user);
+        const optimisedMaximumDate = this.getOptimisedMaximumDate(user);
+        if (this.optimiseTimeline && optimisedMinimumDate != null && optimisedMaximumDate != null) {
+          const tdiff = optimisedMaximumDate - optimisedMinimumDate + window.DAY_IN_MS;
+          const idxs = this.drags.map((x) => (x * tdiff) / 100);
+          const tsince = window.getDateStr(optimisedMinimumDate + idxs[0]);
+          const tuntil = window.getDateStr(optimisedMinimumDate + idxs[1]);
+          this.drags = [];
+          this.openTabZoom(user, tsince, tuntil, isMerged);
+        } else {
+          // additional day was added to include the date represented by filterUntilDate
+          const tdiff = (new Date(this.filterUntilDate)).valueOf() - (new Date(this.filterSinceDate)).valueOf()
+              + window.DAY_IN_MS;
+          const idxs = this.drags.map((x) => (x * tdiff) / 100);
+          const tsince = window.getDateStr(new Date(this.filterSinceDate).getTime() + idxs[0]);
+          const tuntil = window.getDateStr(new Date(this.filterSinceDate).getTime() + idxs[1]);
+          this.drags = [];
+          this.openTabZoom(user, tsince, tuntil, isMerged);
+        }
```

Comment:
This part also has a lot of duplicated calculation logic. If possible, define the `since` and `until` dates locally and use them in common calculation logic.